README.rst

```diff
@@ -72,3 +67,219 @@ response, e.g.
 .. code:: python
 
    WAGTAIL_STORAGES_DOCUMENT_HOOK_ORDER = 900
+
+
+Recommended S3 setup with Wagtail
+---------------------------------
+
+This is a guide that explains the recommended setup for using S3 with Wagtail.
+This guide assumes that:
+
+* You serve your main website at ``llamasavers.com``.
+* Your S3 bucket is called ``media.llamasavers.com`` and you host it from that
+  domain name.
+* Using CDN on that domain, this guide will assume Cloudflare.
+
+Set up S3 bucket
+~~~~~~~~~~~~~~~~
+
+First step is to set up your S3 bucket. It must be configured to:
+
+* Bucket name must match the domain name, e.g. ``media.llamasavers.com``, that
+  you intend using it with.
+* Allow user to perform the following actions on the bucket:
+  * ``s3:ListBucket``
+  * ``s3:GetBucketLocation``
+  * ``s3:ListBucketMultipartUploads``
+  * ``s3:ListBucketVersions``
+* Allow user to perform all the actions (``s3:*``) on the objects within the
+  bucket.
+* Allow the internet traffic to access Wagtail image renditions (``images/*``).
+
+The user permissions can be set in the IAM or via a bucket policy. See example
+of all of those points being achieved with a bucket policy below.
+
+.. code:: json
+
+   {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Sid": "PublicGetObject",
+                "Effect": "Allow",
+                "Principal": "*",
+                "Action": "s3:GetObject",
+                "Resource": "arn:aws:s3:::[BUCKET NAME]/images/*"
+            },
+            {
+                "Sid": "AllowUserManageBucket",
+                "Effect": "Allow",
+                "Principal": {
+                    "AWS": "arn:aws:iam::[USER ARN]"
+                },
+                "Action": [
+                    "s3:ListBucket",
+                    "s3:GetBucketLocation",
+                    "s3:ListBucketMultipartUploads",
+                    "s3:ListBucketVersions"
+                ],
+                "Resource": "arn:aws:s3:::[BUCKET NAME]"
+            },
+            {
+                "Sid": "AllowUserManageBucketObjects",
+                "Effect": "Allow",
+                "Principal": {
+                    "AWS": "arn:aws:iam::[USER ARN]"
+                },
+                "Action": "s3:*",
+                "Resource": "arn:aws:s3:::[BUCKET NAME]/*"
+            }
+        ]
+    }
+
+
+After the bucket is set up, the bucket can be configured in a Wagtail project.
+
+Set up django-storages
+~~~~~~~~~~~~~~~~~~~~~~
+
+You need to install ``django-storages`` and ``boto3``.
+
+.. code:: sh
+
+   pip install django-storages[boto3]
+
+Set up your S3 bucket with ``django-storages``. The following code will allow
+configuration with environment variables.
```

Comment:
"The following code will allow configuration with environment variables." → "The following code allows configuration via environment variables."